From e386e7b14b139f15205e14b173e8222bf38d9e18 Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Wed, 15 Jul 2009 09:43:04 -0700
Subject: [PATCH 2/7] Reset framebuffer offset when rebinding aperture (22760).

scrn->fbOffset may be changed when binding objects to the aperture during
server initialization or VT enter. This was accidentally removed when the
NoAlloc option was eliminated.

Signed-off-by: Keith Packard <keithp@keithp.com>
---
 src/i830_memory.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/i830_memory.c b/src/i830_memory.c
index f2f3966..cc9d376 100644
--- a/src/i830_memory.c
+++ b/src/i830_memory.c
@@ -1581,6 +1581,9 @@ i830_bind_all_memory(ScrnInfoPtr pScrn)
 	i830_update_cursor_offsets(pScrn);
     i830_set_max_gtt_map_size(pScrn);
 
+    if (pI830->front_buffer)
+	pScrn->fbOffset = pI830->front_buffer->offset;
+
     return TRUE;
 }
 
-- 
1.6.3.3

